community/kernel/src/main/java/org/neo4j/kernel/impl/store/TimeZoneMapping.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2002-2018 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.store;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class TimeZoneMapping
+{
+    /**
+     * Prevent instance creation.
+     */
+    private TimeZoneMapping()
+    {
+    }
+
+    private static final List<String> TIME_ZONE_SHORT_TO_STRING = new ArrayList<>( 1024 );
+    private static final Map<String,Short> TIME_ZONE_STRING_TO_SHORT = new HashMap<>( 1024 );
```

Comment:
I'd like to change `TIME_ZONE_SHORT_TO_STRING` to just be an array, and I think we can introduce a lower overhead data structure for `TIME_ZONE_STRING_TO_SHORT`, but those are future improvements that can be made without breaking any functionality.